copilot-language/src/Copilot/Language/Stream.hs

```diff
@@ -29,28 +29,28 @@ import qualified Prelude as P
 -- 'Op2', 'Op3').
 
 data Stream :: * -> * where
-  Append      :: Typed a
+  Append      :: (HasCallStack, Typed a)
               => [a] -> Maybe (Stream Bool) -> Stream a -> Stream a
-  Const       :: Typed a => a -> Stream a
-  Drop        :: Typed a
+  Const       :: (HasCallStack, Typed a) => a -> Stream a
+  Drop        :: (HasCallStack, Typed a)
               => Int -> Stream a -> Stream a
-  Extern      :: Typed a
+  Extern      :: (HasCallStack, Typed a)
               => String -> Maybe [a] -> Stream a
-  Local       :: (Typed a, Typed b)
+  Local       :: (HasCallStack, Typed a, Typed b)
               => Stream a -> (Stream a -> Stream b) -> Stream b
-  Var         :: Typed a
+  Var         :: (HasCallStack, Typed a)
               => String -> Stream a
-  Op1         :: (Typed a, Typed b)
+  Op1         :: (HasCallStack, Typed a, Typed b)
               => Core.Op1 a b -> Stream a -> Stream b
-  Op2         :: (Typed a, Typed b, Typed c)
+  Op2         :: (HasCallStack, Typed a, Typed b, Typed c)
               => Core.Op2 a b c -> Stream a -> Stream b -> Stream c
-  Op3         :: (Typed a, Typed b, Typed c, Typed d)
+  Op3         :: (HasCallStack, Typed a, Typed b, Typed c, Typed d)
               => Core.Op3 a b c d -> Stream a -> Stream b -> Stream c -> Stream d
-  Label       :: Typed a => String -> Stream a -> Stream a
+  Label       :: (HasCallStack, Typed a) => String -> Stream a -> Stream a
```

Comment:
If we want to be very thorough and ensure that every use of these `Stream` constructors have call stack information at their use sites, we could convert these implicit `HasCallStack` constraints into explicit `CallStack` values. This largely comes down to personal taste, however, so feel free to keep the code like this if you prefer it this way.